usage/camp/src/main/java/io/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -122,14 +124,16 @@ public Entity call() throws Exception {
                 default:
                     throw new IllegalStateException("Unexpected scope "+scope);
             }
+            Predicate<Entity> configEqualTo = EntityPredicates.configEqualTo(BrooklynCampConstants.PLAN_ID, componentId);
+            Optional<Entity> result = Iterables.tryFind(entitiesToSearch, configEqualTo);
+            if (result.isPresent()) return result.get();
             
-            Optional<Entity> result = Iterables.tryFind(entitiesToSearch, EntityPredicates.configEqualTo(BrooklynCampConstants.PLAN_ID, componentId));
-            
-            if (result.isPresent())
-                return result.get();
+            Predicate<Entity> typeNameEquals = EntityPredicates.typeNameEquals(componentId);
```

Comment:
Would be good to have a unit test for this. For example, see `EntitiesYamlTest.testComponent`. We should also have a test for the id lookup approach with `$brooklyn:descendant("myid")`.